deploy/sre-prometheus/100-control-plane-resizing.PrometheusRule.yaml

```diff
@@ -0,0 +1,56 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  labels:
+    prometheus: sre-control-plane-resizing-alerts
+    role: alert-rules
+  name: sre-control-plane-resizing-alerts
+  namespace: openshift-monitoring
+spec:
+  # https://issues.redhat.com/browse/OSD-3215
+  # https://docs.openshift.com/container-platform/4.3/scalability_and_performance/recommended-host-practices.html#master-node-sizing_
+  groups:
+  - name: sre-control-plane-resizing-recording.rules
+    rules:
+      - expr: label_replace(cluster:nodes_roles, "instance", "$1", "node", "(.*)") * on(instance) group_left node_memory_MemTotal_bytes
+        record: sre:node_roles:memory_total_bytes
+      - expr: count(cluster:nodes_roles{label_node_role_kubernetes_io!~"(master|infra)"})
+        record: sre:node_workers:count
+      # master has less than 16GB RAM, worker count > 25, and worker count <= 100
+      # master has less than 32GB RAM, worker count > 100, and worker count <= 250
+      # master has less than 64GB RAM, worker count > 250
+      - expr: (
+                avg(sre:node_roles:memory_total_bytes{label_node_role_kubernetes_io="master"}) < 16*1024*1024*1024
+                AND sre:node_workers:count > 25 
+                AND sre:node_workers:count <= 100 
+              )
+              OR 
+              (
+                avg(sre:node_roles:memory_total_bytes{label_node_role_kubernetes_io="master"}) < 32*1024*1024*1024
+                AND sre:node_workers:count > 100
+                AND sre:node_workers:count <= 250 
+              )
+              OR 
+              (
+                avg(sre:node_roles:memory_total_bytes{label_node_role_kubernetes_io="master"}) < 64*1024*1024*1024
+                AND sre:node_workers:count > 250
+              )
+        record: sre:node_masters:need_resize
+  - name: sre-control-plane-resizing-alerts
+    rules:
+      - alert: MasterNodesNeedResizingSRE
+        expr: sre:node_masters:need_resize > 0
+        for: 15m
+        labels:
+          severity: warning
+          namespace: openshift-monitoring
+        annotations:
+          message: "The cluster's master instance has been undersized for 15 minutes and should be vertically scaled to support the existing workers.  See linked SOP for details.  Critical alert will be raised at 2 hours."
+      - alert: MasterNodesNeedResizingSRE
+        expr: sre:node_masters:need_resize > 0
+        for: 2h
+        labels:
+          severity: critical
+          namespace: openshift-monitoring
+        annotations:
+          message: "The cluster's master instance has been undersized for 2 hours and must be vertically scaled to support the existing workers.  See linked SOP for details."
```

Comment:
Where are we linking an SOP for this? Is it still TBD?

Reply:
Auto-generated based on alert name: https://github.com/openshift/ops-sop/blob/master/v4/alerts/MasterNodesNeedResizingSRE.md

Reply:
Non blocking comment but how does that auto gen work?